### PR TITLE
Update consul plugin to use consul-snapshot

### PR DESCRIPTION
**Pull request by bodymindarts**
Since Consul 0.7.1 the command `consul snapshot` has been available to backup and restore all of Consuls entities to a local file.
This PR updates the consul plugin to make use of the native consul snapshot command for backup and restore.

### CODE REVIEW

File: plugin/consul/plugin.go [hunk 1]
```diff
@@ -42,109 +43,153 @@
...
+func (p ConsulPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
 	var (
 		s    string
-		b    bool
 		err  error
 		fail bool
 	)
 
-	s, err = endpoint.StringValueDefault("host", "")
+	s, err = endpoint.StringValueDefault("consul", "")
```
Comment (jhunt) [inline plugin/consul/plugin.go]: Is this a backwards-breaking change?  Will existing configurations with a `host` key continue to work once this code is merged in?

File: plugin/consul/plugin.go [hunk 2]
```diff
@@ -42,109 +43,153 @@
...
 	} else if s == "" {
-		ansi.Printf("@G{\u2717 host                  using default host @C{%s}}\n", DefaultHostPort)
+		ansi.Printf("@G{\u2713 consul}       using default consul @C{%s}\n", DefaultConsul)
 	} else {
-		ansi.Printf("@G{\u2713 host}                  @C{%s}\n", s)
+		ansi.Printf("@G{\u2713 consul}       @C{%s}\n", s)
 	}
 
-	b, err = endpoint.BooleanValueDefault("skip_ssl_validation", false)
+	s, err = endpoint.StringValueDefault("address", "")
```
Comment (jhunt) [inline plugin/consul/plugin.go]: Is this also a backwards-breaking change?

**Comment by bodymindarts** [comment]
You are right this is backwards incompatible. I'll add this code as a new plugin `consul-snapshot` and add a depreciation warning to the old one.

**Comment by jhunt** [comment]
Why not just add a `use_snapshot` option to this plugin?

**Comment by bodymindarts** [comment]
To keep the code cleaner.
The 2 plugins have different options and different mechanisms of execution and different outcomes.
Both from the code author and end-users perspective they are different.
I don't see a benefit of merging together these 2 plugins with a bunch of conditional branching code. Will make the code much less understandable.

**Comment by jhunt** [comment]
A good justification.

I will go ahead and merge this PR.

**Pull request merged**